server/controllers/metrics.js

```diff
@@ -0,0 +1,34 @@
+import campaignsService from '../db/services/campaigns';
+import scriptsService from '../db/services/scripts';
+
+import { checkCampaignNotDraft, countStatusAndOutcomeFrequency, handleQuestionResponses, mapCallIds } from './controller_helpers/metrics';
+
+export default function getCampaignMetrics(req, res) {
+  const metrics = {
+    status_distribution: {},
+    outcome_distribution: {},
+    response_stats: {}
+  };
+  const campaignId = parseInt(req.params.id, 10);
+  try {
+    checkCampaignNotDraft(campaignId, res);
+  } catch (e) {
+    return res.status(400).json({ message: e });
```

Comment:
A 204 might be more appropriate unless we want client-side logic to ban metric queries for draft campaigns.

Reply:
Yep, this is changed